src/core/reopt.jl

```diff
@@ -422,16 +424,37 @@ function build_reopt!(m::JuMP.AbstractModel, p::REoptInputs)
 		add_to_expression!(Costs, m[:Lifecycle_Emissions_Cost_Health])
 	end
 	
-	@objective(m, Min, m[:Costs])
-	
-	if !(isempty(p.s.storage.types.elec)) && p.s.settings.add_soc_incentive # Keep SOC high
-		@objective(m, Min, m[:Costs] - 
-		sum(m[:dvStoredEnergy][b, ts] for b in p.s.storage.types.elec, ts in p.time_steps) /
-			(8760. / p.hours_per_time_step)
+	@expression(m, Objective,
+		m[:Costs]
+	)
+		
+	if !(isempty(p.s.storage.types.elec)) && p.s.settings.add_soc_incentive
+		# Incentive to keep SOC high
+		add_to_expression!(
+			Objective, 
+			- sum(
+				m[:dvStoredEnergy][b, ts] for b in p.s.storage.types.elec, ts in p.time_steps
+			) / (8760. / p.hours_per_time_step)
+		)
+	end
+	if !isempty(p.s.electric_utility.outage_durations)
+		# Incentive to minimize unserved load in each outage, not just the max over outage start times
+		add_to_expression!(
+			Objective, 
+			sum(sum(0.0001 * m[:dvUnservedLoad][s, tz, ts] for ts in 1:p.s.electric_utility.outage_durations[s]) for s in p.s.electric_utility.scenarios, tz in p.s.electric_utility.outage_start_time_steps)
```

Comment:
We don't want to use the value of lost load here because that is already included in the objective (see [core/reopt.jl line 414](https://github.com/NREL/REopt.jl/blob/outages/src/core/reopt.jl#L416) and https://github.com/NREL/REopt.jl/blob/outages/src/constraints/outage_constraints.jl#L53). Across given outage start time steps, only the worst case outage cost is added to the objective (i.e. we are already doing your option 1). That means that in the non-worst outage scenarios, the model has no disincentive to drop load up to equal cost to the worst outage. This wouldn't matter except that the results for these non-worst outage scenarios are also reported. These results may not show the minimum cost achievable for those non-worst outage scenarios and can even vary run-to-run for the same inputs. I added the incentive here to address that. So, in the sense that 0.001 is an arbitrary tiny value to incentive a certain behavior in the solution, this is more like the latter case you described, just not for the specific purpose you described. The value is meant to be negligible relative to any other incentives in the optimization, only differentiating between otherwise equal cost solutions. It just needs to be large enough to be significant considering optimality tolerance settings.
Side note, we use expected values calculated over outage duration. The user can provide multiple outage durations and their respective probabilities. So the outage cost added to the objective is actually an expected value in one dimension (duration) and a max (so robust optimization) in the other dimension (start time).
Does that clear things up? Do you still think something should change?

Reply:
I see now - for each outage duration we pick the worst possible start time, and then perform a probability-weighted sum of those outages to calculate expected outage cost.  This makes sense, thanks a bunch for clarifying.